src/interfaces/book.ts

```diff
@@ -7,3 +7,14 @@ export interface Book {
   year: string;
   imageUrl: string | null;
 }
+
+export interface BookState {
+  books: Book[];
+  booksLoading: boolean;
+  booksError: string | null;
+}
+
+export interface BookAction {
+  type: string;
+  payload: any;
```

Comment:
Type this interface: `payload: Book[] | string;` => `Book[]` for `GET_BOOKS_SUCCESS` and `string` for `GET_BOOKS_FAILURE`